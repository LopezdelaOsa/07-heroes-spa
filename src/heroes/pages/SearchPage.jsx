
import { useLocation, useNavigate } from 'react-router-dom'
import { HeroCard} from '../components'
import { useForm } from '../hooks/useForm'
import queryString from 'query-string' // yarn add query-string
import { getHeroByName } from '../helpers'

export const SearchPage = () => {

  const navegar = useNavigate()
  const localizacion = useLocation()
  //console.log(localizacion)
  const {q = ''} = queryString.parse(location.search)
  //console.log(query)
  const heroes = getHeroByName(q)

  const muestraBusqueda = (q.length === 0)
  const muestraError = (q.length > 0) && heroes.length === 0

  const {textoBusqueda, onInputChange } = useForm({
    textoBusqueda: q
  })

  const onEnviarBusqueda = (event) => {
    event.preventDefault()
    //if ( textoBusqueda.trim().length <= 1 ) return

    navegar(`?q=${textoBusqueda}`)

  }

    return (
      <>
        <h1>Búsquedas...</h1>
        <hr/>

        <div className="row">
          <div className="col-5">
            <h4>Buscando</h4>
            <hr/>
            <form onSubmit={onEnviarBusqueda}>
              <input type="text" placeholder="Buscar un héroe" className="form-control" autoComplete="off"
                name="textoBusqueda"
                value={textoBusqueda}
                onChange={onInputChange}
              />
              <button className="btn btn-outline-primary mt-1">
                Buscar
              </button>


            </form>
          </div>

          <div className="col-7">
          <h4>Resultados</h4>
          <hr/>
            {/* {
              ( q === '' ) 
              ? <div className="alert alert-primary">Héroes...</div>
              : (heroes.length === 0) && <div className="alert alert-danger">No hay coincidencias con <b>{ q }</b></div>
            } */}
            <div className="alert alert-primary animate__animated animate__fadeIn" style={{ display: muestraBusqueda ? '' : 'none' }}>
              Héroes...
            </div>

            <div className="alert alert-danger animate__animated animate__fadeIn" style={{ display: muestraError ? '' : 'none' }}>
              No hay coincidencias con <b>{ q }</b>
            </div>

            {/* <HeroCard {...hero} /> */}
            {
              heroes.map (hero => (
                <HeroCard key={ hero.id } {...hero} />
              ))
            }

          </div>

        </div>



      </>
      
    )
  }