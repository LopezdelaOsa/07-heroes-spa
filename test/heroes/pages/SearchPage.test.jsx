import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SearchPage } from "../../../src/heroes/pages/SearchPage"

describe('Pruebas en <SearchPage/>', () => {

    test('Debe mostrar correctamente con valores por defecto', () => {
        const {contanier} = render(
            <MemoryRouter>
                <SearchPage/>
            </MemoryRouter>
        )

        expect(contanier).toMatchSnapshot()
        //screen.debug()
    })


    test('Debe mostrar a Batman y el input con el valor del querystring', () => {
        render(
            //<MemoryRouter initialEntries={['/search?q=batman']}>
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>
        )
        //screen.debug()
        const input = screen.getByRole('textbox')
        expect (input.value).toBe('batman')

        const img = screen.getByRole('img')
        expect (img.src).toContain('/assets/heroes/dc-batman.jpg')

        const alertD = screen.getByLabelText('alert-danger')
        expect (alertD.style.display).toBe('none')
        //screen.debug()
    })

})