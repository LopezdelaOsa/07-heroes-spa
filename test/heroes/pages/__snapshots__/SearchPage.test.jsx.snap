// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage/> Debe mostrar correctamente con valores por defecto 1`] = `undefined`;
